services/jenkins/jenkins-coverage.service.js

```diff
@@ -1,55 +1,60 @@
 'use strict'
 
 const Joi = require('joi')
-const { BaseJsonService } = require('..')
-const serverSecrets = require('../../lib/server-secrets')
+const { coveragePercentage } = require('../color-formatters')
+const JenkinsBase = require('./jenkins-base')
 const {
-  coveragePercentage: coveragePercentageColor,
-} = require('../color-formatters')
-
-const jacocoCoverageSchema = Joi.object({
-  instructionCoverage: Joi.object({
-    percentage: Joi.number()
-      .min(0)
-      .max(100)
-      .required(),
-  }).required(),
-}).required()
-
-const coberturaCoverageSchema = Joi.object({
-  results: Joi.object({
-    elements: Joi.array()
-      .items(
-        Joi.object({
-          name: Joi.string().required(),
-          ratio: Joi.number()
-            .min(0)
-            .max(100)
-            .required(),
-        })
+  buildTreeParamQueryString,
+  buildUrl,
+  queryParamSchema,
+} = require('./jenkins-common')
+
+const formatMap = {
+  jacoco: {
+    schema: Joi.object({
+      instructionCoverage: Joi.object({
+        percentage: Joi.number()
+          .min(0)
+          .max(100)
+          .required(),
+      }).required(),
+    }).required(),
+    treeQueryParam: 'instructionCoverage[percentage]',
+    transform: json => ({ coverage: json.instructionCoverage.percentage }),
+  },
+  cobertura: {
+    schema: Joi.object({
+      results: Joi.object({
+        elements: Joi.array()
+          .items(
+            Joi.object({
+              name: Joi.string().required(),
+              ratio: Joi.number()
+                .min(0)
+                .max(100)
+                .required(),
+            })
+          )
+          .has(Joi.object({ name: 'Lines' }))
+          .min(1)
+          .required(),
```

Comment:
Jenkins will return an HTTP 404 if a job is requested that did not publish code coverage in the cobertura format (which is why on a 404 we use the error message `job or coverage not found`). 

If the target job has cobertura coverage then the elements array will include an object with a `Lines` field, which is what we need.